src/Core/Settings.cpp

```diff
@@ -358,9 +358,12 @@ Each time this number of parts was uploaded to S3, s3_min_upload_part_size is mu
 )", 0) \
     DECLARE(UInt64, s3_max_part_number, S3::DEFAULT_MAX_PART_NUMBER, R"(
 Maximum part number number for s3 upload part.
+)", 0) \
+    DECLARE(Bool, s3_allow_multipart_copy, true, R"(
```

Comment:
This setting was added because for some cloud storages they want to always use single-operation copy.

Reply:
Alibaba Cloud has a logic where if the source object and target location are in the same bucket, it creates a symbolic link instead of copying the entire object. Thus copying of a single object is always very fast regardless of its size and they don't to use multipart copy at all (https://github.com/ClickHouse/support-escalation/issues/3763#issuecomment-2652483289)